### PR TITLE
[build] Build better-sqlite3 on windows during Jenkins CI

### PR DESCRIPTION
**Pull request by johnbotris**
In order to fix the windows build we needed to also build the better-sqlite3 native module in windows. We had also inadvertently broken the keytar build step too. I refactored a bit to use `nativeLibraryProvider.js` as a script for building libs on demand, and allowing them to be easily consumed by our rollup plugin

### CODE REVIEW

**Comment by johnbotris** [comment]
One point to resolve is:

Is it ok to just bump down our node-gyp version to 6.1, or do we want to just switch versions when building better-sqlite3 in windows

File: buildSrc/buildUtils.js [hunk 1]
```diff
@@ -12,24 +13,30 @@ var measureStartTime
...
 }
 
-async function loadPackageJson() {
-	return JSON.parse(await fs.readFile(path.join(__dirname, "..", "package.json"), "utf8"))
+/**
+ * Get the installed version of a module
+ * @param module {string}
+ * @returns {string}
+ */
+export function getInstalledModuleVersion(module) {
```
Comment (charlag) [inline buildSrc/buildUtils.js:36]: Need to re-do with #3924

File: buildSrc/nativeLibraryProvider.js [hunk 2]
```diff
@@ -1,95 +1,308 @@
...
+					environment === "electron"
+						? [
+							"--runtime=electron",
+							'--dist-url=https://www.electronjs.org/headers',
+							`--target=${getElectronVersion()}`,
+						]
+						: []
+				)
+			],
+			cwd: path.join(rootDir, "node_modules", nodeModule),
```
Comment (charlag) [inline buildSrc/nativeLibraryProvider.js:169]: Redo this with #3924

File: buildSrc/nativeLibraryProvider.js [hunk 3]
```diff
@@ -1,95 +1,308 @@
...
-						`--target=${electronVersion}`,
+						`--runtime=${target.runtime}`,
+						`--target=${target.version}`,
 					]
 					: []
-			)
+			),
+			"--verbose"
 		],
+		cwd: path.join(rootDir, "node_modules", nodeModule),
```
Comment (charlag) [inline buildSrc/nativeLibraryProvider.js:221]: This CWD too

File: buildSrc/signDesktopClients.js [hunk 4]
```diff
@@ -6,6 +6,7 @@
 import fs from "fs-extra"
 import path from "path"
 import options from "commander"
+import {fileExists} from "./buildUtils"
```
Comment (charlag) [inline buildSrc/signDesktopClients.js:9]: Borked

**Review by charlag — APPROVED** [review]
Should re-do some things after merge

**Pull request merged**